### PR TITLE
fix cuda OOM by using single Prior

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Fixes the OOM issue with the `5b`  model and `fp16` sampling. 
Also fixes the slow generation test by sending each prior to `cuda` only when they are actually used.

### CODE REVIEW

**Comment by ArthurZucker** [comment]
This will require #20485 to be merged, otherwise the logits for `fp16_sampling` will differ.

**Comment by ydshieh** [comment]
I will  wait the mentioned PR #20485 :-) then back to this one. Also see my comment in that PR 🙏 @ArthurZucker 🙏

**Comment by ArthurZucker** [comment]
I'll just push to model to another repo and ping you back here!

File: tests/models/jukebox/test_modeling_jukebox.py [hunk 1]
```diff
@@ -335,10 +335,11 @@ def test_slow_sampling(self):
 
     @slow
     def test_fp16_slow_sampling(self):
-        model = JukeboxModel.from_pretrained(self.model_id, min_duration=0).eval().half().to("cuda")
-        labels = [i.cuda() for i in self.prepare_inputs(self.model_id)]
+        prior_id = "ArthurZ/jukebox_prior_0"
```
Comment (ydshieh) [inline tests/models/jukebox/test_modeling_jukebox.py:338]: This PR change the test from `JukeboxModel` to `JukeboxPrior`. I am not very familiar with Jukebox, probably this is good enough. But maybe if we can create a `JukeboxModel` with a single `JukeboxPrior` (that is from your `"ArthurZ/jukebox_prior_0"`), and test against `JukeboxModel`, it would be even better.

I leave you to make the final call @ArthurZucker

**Review by ydshieh — APPROVED** [review]
LGTM, thank you!

File: tests/models/jukebox/test_modeling_jukebox.py (hunk 1 above)
Comment (ydshieh) [reply tests/models/jukebox/test_modeling_jukebox.py:338]: BTW, is this "ArthurZ/jukebox_prior_0" no longer a real model but instead a randomly initialized model?

Comment (ArthurZucker) [reply tests/models/jukebox/test_modeling_jukebox.py:338]: Will leave it as is for simplicity 😉

**Review by sgugger — APPROVED** [review]
Thanks for fixing it this way!
